src/tabs/tabs.js

```diff
@@ -152,6 +169,11 @@ angular.module('ui.bootstrap.tabs', [])
         } else {
           scope.index = 0;
         }
+      } else {
+        // watch when index is ng-repeat $index
+        scope.$parent.$watch($parse(attrs.index), function() {
```

Comment:
This needs to garbage collect the watcher when the directive scope is destroyed.